pkg/kubectl/cmd/util/factory_object_mapping.go

```diff
@@ -258,6 +258,13 @@ func (f *ring1Factory) LogsForObject(object, options runtime.Object, timeout tim
 			return nil, fmt.Errorf("invalid label selector: %v", err)
 		}
 
+	case *extensions.DaemonSet:
+		namespace = t.Namespace
+		selector, err = metav1.LabelSelectorAsSelector(t.Spec.Selector)
+		if err != nil {
+			return nil, fmt.Errorf("invalid label selector: %v", err)
+		}
+
```

Comment:
Add this to `func AttachablePodForObject` too?

Reply:
done, good catch